### DISCUSSION TITLE
Labels with rocAL fn.readers.video?

### DISCUSSION THREAD

**Discussion by hbfreed** [ROCm/ROCm#2793] [Q&A]
### I also posted this over on the MIvisionX dicussion board as well, but there's not a whole lot of activity over there, hope that's ok. 

Is it possible to actually have labels with rocAL's fn.readers.video? I have tried a bunch of different things, but just can't get labels out. This is my most recent attempt, with lists for filenames and labels (read in from a .csv). The ROCALVideoIterator is ripped pretty much straight from the video example found in the rocAL examples folder. 

I'm using an EPYC 7402p and a 7900xtx on Ubuntu 22.04.3 LTS. 

Here's the simple pipe that I'm using:

```
fileroot = 'Documents/val'
size = len(os.listdir(file_root))

df = pd.read_csv(val_labels.csv,header=None)
filenames_list, labels_list = df[0].tolist(),df[1].tolist()

pipe = Pipeline(batch_size=2, num_threads=48, seed=10, device_id=0,
            rocal_cpu=False, tensor_dtype = types.FLOAT16,
            prefetch_queue_depth = 20)
with pipe:
    videos = fn.readers.video(file_root=file_root,
                              shard_id=0, 
                              num_shards=1, 
                              random_shuffle=True,
                              sequence_length=130,
                              filenames=filenames_list,
                              labels=labels_list)
pipe.set_outputs(videos)



pipe.build()

train_loader = ROCALVideoIterator(pipe,size)
for i, (inputs,labels) in enumerate(train_loader):
    print(inputs.shape,labels.shape)
```

**Comment by rrawther** [comment]
Yes, we can get the labels as well. I will post the code for getting the labels

**Reply by hbfreed** [reply]
Excellent, thanks a ton! Looking forward to that : )
